src/expression.c

```diff
@@ -10043,6 +10107,39 @@ Expression *SliceExp::semantic(Scope *sc)
         lwr = lwr->optimize(WANTvalue);
         upr = upr->optimize(WANTvalue);
 
+        // avoid slice bounds-checking. TODO use resolveOpDollar?
+        {
+            // lower
+            dinteger_t lwrMul = 1; // non-one means [ $*lwrMul .. _ ]
+            dinteger_t lwrDiv = 1; // non-one means [ $/lwrDiv .. _ ]
+            const Boundness lwrBoundness = this->analyzeRelativeBound(lwr, &lwrMul, &lwrDiv);
+            if (lwrBoundness >= inside)
+            {
+                this->lowerIsInBounds = true;
+            }
+
+            // upper
+            dinteger_t uprMul = 1; // non-one means [ $*uprMul .. _ ]
+            dinteger_t uprDiv = 1; // non-one means [ $/uprDiv .. _ ]
+            const Boundness uprBoundness = this->analyzeRelativeBound(upr, &uprMul, &uprDiv);
+            if (uprBoundness >= inside)
+            {
+                this->upperIsInBounds = true;
+            }
+
+            // lower and upper
+            if ((this->lowerIsInBounds &&
+                 this->upperIsInBounds) &&
+                (lwrBoundness == atStart || // [0 .. _]
```

Comment:
Don't put comments in the middle of expressions, the ddmd converter will choke on this.